src/components/calendar/days-view/days-view.ts

```diff
@@ -481,12 +480,17 @@ export default class IgcDaysViewComponent extends EventEmitterMixin<
     const last = weeks.length - 1;
 
     for (const [idx, week] of weeks.entries()) {
+      const weekDaysProps: any[] = [];
+      for (const day of week) {
+        weekDaysProps.push(this.getDayProperties(day, today));
+      }
+      const hidden = weekDaysProps.every((p) => p.hidden);
```

Comment:
IMO it would be better to check just the first and the last week whether all the days are hidden before yielding the DOM pieces.